runtime/src/bank/tests.rs

```diff
@@ -12823,6 +12823,94 @@ fn test_rewards_computation() {
     assert_eq!(stake_rewards.stake_rewards.len(), expected_num_delegations);
 }
 
+/// Test rewards compuation and partitioned rewards distribution at the epoch boundary
+#[test]
+fn test_rewards_computation_and_partitioned_distribution() {
+    solana_logger::setup();
+
+    // setup the expected number of stake delegations
+    let expected_num_delegations = 100;
+
+    let validator_keypairs = (0..expected_num_delegations)
+        .map(|_| ValidatorVoteKeypairs::new_rand())
+        .collect::<Vec<_>>();
+
+    let GenesisConfigInfo { genesis_config, .. } = create_genesis_config_with_vote_accounts(
+        1_000_000_000,
+        &validator_keypairs,
+        vec![2_000_000_000; expected_num_delegations],
+    );
+
+    let bank0 = Bank::new_for_tests(&genesis_config);
+    let num_slots_in_epoch = bank0.get_slots_in_epoch(bank0.epoch());
+    assert_eq!(num_slots_in_epoch, 32);
+
+    let mut previous_bank = Arc::new(Bank::new_from_parent(
+        &Arc::new(bank0),
+        &Pubkey::default(),
+        1,
+    ));
+
+    // simulate block progress
+    for slot in 1..num_slots_in_epoch + 2 {
+        let pre_cap = previous_bank.capitalization();
+        let curr_slot = slot + 1;
```

Comment:
this would be simpler to me if it was `for curr_slot in 2..=num_slots_in_epoch + 2 {`.
I think `slot` is never used?
or just use `slot` as the name of the variable in the for loop and get rid of `curr_slot`

Reply:
yes. good observation.
`slot` was used before. But after refactoring with review comments, it is no longer used anymore.
updated.